src/index.html

```diff
@@ -6,7 +6,41 @@
     <title>Calendar</title>
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
-  <body>
-    <h1>Calendar</h1>
+  <body class="container">
+
+    <div class="calendar calendar__day--sun calendar__month-length--31">
```

Comment:
```suggestion
    <div class="calendar calendar--day-sun calendar--month-length-31">
```

because block is calendar, you are creating modifier for `calendar`
so {block/element}--{modifier(day-sun in your case)}